designs/0031-deprecation-schedule.md

```diff
@@ -0,0 +1,254 @@
+- Feature Name: deprecation_schedule
+- Start Date: 2021-09-21
+- RFC PR: (leave this empty)
+- Stan Issue: (leave this empty)
+
+# Summary
```

Comment:
The document needs a more descriptive title than "summary".

Reply:
I was just going off the template in this repo. It seems like other docs all use the same format? E.g. https://github.com/stan-dev/design-docs/blob/master/designs/0003-opencl_kernel_generator.md#summary

Reply:
I'd prefer a title, but will accept "Summary" given it's what the ones that have been merged do (other than CmdStanPy).